app/cli/cmd/registered_integration_add.go

```diff
@@ -66,5 +123,39 @@ func parseKeyValOpts(opts []string) (map[string]any, error) {
 		}
 		options[kv[0]] = kv[1]
 	}
+
+	// 2 - Cast the values to the expected type defined in the schema
+	for k, v := range options {
+		prop, ok := propertiesMap[k]
+		if !ok {
+			continue
+		}
+
+		switch prop.Type {
+		case "string":
+			options[k] = v.(string)
+		case "integer":
```

Comment:
ok, that's what we are missing. I will update smtp/port implementation once you land this diff.